app/views/pages/home.html.erb

```diff
@@ -1,3 +1,17 @@
+<%= form_for  :filter, url: root_path, method: :get, local: false do |f|%>
+  <div class="row">
+    <div class="col">
+      <%= f.label :category_id, "Category" %>
+      <%= f.select :category_id, @categories.pluck(:name, :id), class: "form-control"%>
```

Comment:
dlaczego wywolywanie metod activerecord na widoku jest niedobre?

Reply:
Separation of concerns. 
Zrobiłem w modelu coś takiego: 
```
def self.select_params
    pluck(:name, :id)
  end
```

controller:
`@categories_select_params = Category.select_params`

więc widok wygląda teraz tak:
` <%= f.select :category_id, @categories_select_params, class: "form-control"%>`

ale zastanawiam się bo gdybym w kontrolerze miał `@categories` bo potrzebowałbym np `category.image` to chyba lepiej jest w kontrolerze zrobić 
```
@categories = Category.all
@categories_select_params = @categories.pluck(:name, :id) 
```
bo jest tylko jedno zapytanie do db.

Reply:
eleganko. Jest jeszcze pattern, ze zwraca sie presenter do widoku
```
render :new, locals: { presenter: ProductsPresenter.new }
```

gdzie ten presenter zwraca juz kolekcje dla selecta :v: